src/core/data_transform/transforms.py

```
@@ -61,6 +61,8 @@ class PointCloudFusion(object):
     """
 
     def _process(self, data_list):
+        if len(data_list) == 0:
+            return Data()
```

Are you sure ? Won't it be better to raise an Exception there ?

---

I think it's ok, it is happening on on degenerated and if you try to do something useful with what it returns you'll quickly see that something is wrong (data is empty)